fhir-validation/src/main/java/com/ibm/watson/health/fhir/validation/FHIRValidator.java

```diff
@@ -36,44 +34,39 @@
 public class FHIRValidator {
     public static boolean DEBUG = false;
     
-    private final FHIRPathTree tree;
+    private ValidatingVisitor visitor = new ValidatingVisitor();
     
-    private FHIRValidator(FHIRPathTree tree) {
-        this.tree = Objects.requireNonNull(tree);
-        if (!this.tree.getRoot().isResourceNode()) {
-            throw new IllegalStateException("Root must be a resource node");
-        }
-    }
+    private FHIRValidator() { }
     
-    public List<Issue> validate() throws FHIRValidationException {
+    public List<Issue> validate(Resource resource) throws FHIRValidationException {
+        Objects.requireNonNull(resource);
         try {
-            ValidatingVisitor visitor = new ValidatingVisitor(tree);
-            tree.getRoot().asResourceNode().resource().accept(visitor);
-            return visitor.getIssues();
+            return visitor.validate(resource);
         } catch (Exception e) {
             throw new FHIRValidationException("An error occurred during validation", e);
         }
     }
     
-    public static FHIRValidator validator(FHIRPathTree tree) {
-        return new FHIRValidator(tree);
-    }
-    
-    public static FHIRValidator validator(Resource resource) {
-        return new FHIRValidator(FHIRPathTree.tree(resource));
+    public static FHIRValidator validator() {
```

Comment:
doesn't seem to be a singleton pattern, so why not just let them new it themselves at this point?

Reply:
3 reasons:
1. I want to control construction
2. method chaining is a little cleaner looking
3. it follows the patterns used everywhere else in the code base